space-cli/cmd/modules/operations/destroy.go

```diff
@@ -22,55 +22,69 @@ func Destroy() error {
 	}
 
 	// get all containers containing < space-cloud > in their name
-	args := filters.Arg("name", "space-cloud")
-	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{Filters: filters.NewArgs(args), All: true})
+	argsName := filters.Arg("name", "space-cloud")
+	argsNetwork := filters.Arg("network", utils.GetNetworkName(clusterID))
+	containers, err := cli.ContainerList(ctx, types.ContainerListOptions{Filters: filters.NewArgs(argsNetwork, argsName), All: true})
 	if err != nil {
 		_ = utils.LogError(fmt.Sprintf("Unable to list containers - %s", err.Error()), nil)
 		return err
 	}
 
 	// Remove all container
 	for _, containerInfo := range containers {
-		// remove the container from host machine
 		if err := cli.ContainerRemove(ctx, containerInfo.ID, types.ContainerRemoveOptions{Force: true}); err != nil {
 			_ = utils.LogError(fmt.Sprintf("Unable to remove container %s - %s", containerInfo.ID, err.Error()), nil)
 			return err
 		}
 	}
 
 	// Remove the space-cloud network
+	args := filters.Arg("name", utils.GetNetworkName(clusterID))
 	nws, err := cli.NetworkList(ctx, types.NetworkListOptions{Filters: filters.NewArgs(args)})
 	if err != nil {
 		return utils.LogError("Unable to list networks", err)
 	}
 	for _, nw := range nws {
-		_ = cli.NetworkRemove(ctx, nw.ID)
+		if nw.Name == utils.GetNetworkName(clusterID) {
```

Comment:
do we need to do this, as we already using the filter?

Reply:
yes. i tried the normal way but i received multiple network